client/src/pages/Configuration/Configuration.tsx

```diff
@@ -0,0 +1,141 @@
+import { Box, Button, Checkbox, Group, NumberInput, Paper, Select, Space, TextInput } from '@mantine/core';
+import axios from 'axios';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { Logo } from '../../components/Logo';
+import { Paths } from '../../routes/paths';
+import { Header } from './Header/Header';
+import { InfoCircle } from './Info/Info';
+import { Rules } from './Rules/Rules';
+
+export const Configuration = () => {
+  const [nickname, setNickname] = useState('');
+  const [contractType, setContractType] = useState<string | null>(null);
+  const [toWorkDistance, setToWorkDistance] = useState<number | undefined>(2);
+
+  const [acceptRules, setAcceptRules] = useState(false);
+  const [dataProccesingAgreement, setDataProccesingAgreement] = useState(false);
+
+  const navigate = useNavigate();
+
+  const handleSubmit = async () => {
+    const mandatoryInfo = { nickname, contractType, toWorkDistance };
+
+    await axios.post(`${process.env.VITE_SERVER_URL}/api/users/`, mandatoryInfo);
+    navigate(Paths.Home);
```

Comment:
To dobrze by było w try catch zamknąć i przy błędzie jakąś notyfikacje/toast pokazać użytkownikowi